src/dc_receive_imf.rs

```diff
@@ -4933,4 +4936,56 @@ Message with references."#;
 
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_long_filenames() -> Result<()> {
+        let mut tcm = TestContextManager::new().await;
+        let alice = tcm.alice().await;
+        let bob = tcm.bob().await;
+
+        for filename_sent in &[
+            "foo.bar very long file name test baz.pdf",
+            "foobarabababababababbababababverylongfilenametestbaz.pdf",
+            "fooo...pdf",
+            "foo. .pdf",
```

Comment:
`aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.tar.gz` and `a.tar.gz` would be good files to add as testcases.  `a.a..a.a.a.a.tar.gz` if you really want to go wild.